packages/scandipwa/src/component/ProductWishlistButton/ProductWishlistButton.container.js

```diff
@@ -108,23 +109,35 @@ export class ProductWishlistButtonContainer extends PureComponent {
             wishlistId
         } = this.props;
 
+        const {
+            isWishListToggle
+        } = this.state;
+
         if (!isSignedIn()) {
             return showNotification('info', __('You must login or register to add items to your wishlist.'));
         }
 
         this.setWishlistButtonLoading(true);
+        
+        if (!isWishListToggle) {
+            this.setState({ isWishListToggle: true });
+        } else {
+            return;
+        }
 
         if (add) {
             await addProductToWishlist({
                 items: magentoProduct,
                 wishlistId
-            });
+            }).then(() => this.setState({ isWishListToggle: false }));
```

Comment:
Why then? You anyway awaiting for addProductToWishlist, if you will put state setting after it I guess result will be same

Reply:
I understand this, I didn't notice it when refactor code